conda/trust/signature_verification.py

```diff
@@ -217,23 +217,21 @@ def _fetch_channel_signing_data(self, signing_data_url, filename, etag=None, mod
             # TODO: additional loading and error handling improvements?
             raise ValueError(f"Invalid JSON returned from {signing_data_url}/{filename}")
 
-    def __call__(self, info, fn, signatures):
-        if not self.enabled or fn not in signatures:
+    def __call__(self, record):
+        if not self.enabled or not hasattr(record, 'signatures') or not record.signatures:
```

Comment:
good catch, I'd like to avoid any `hasattr` calls, I consider it to be an anti-pattern:

```suggestion
        if not self.enabled or not getattr(record, 'signatures', None):
```